homeassistant/components/homewizard/number.py

```diff
@@ -0,0 +1,69 @@
+"""Creates HomeWizard Number entities."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up numbers for device."""
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if coordinator.data["state"]:
+        async_add_entities(
+            [
+                HWEnergyNumberEntity(coordinator, entry),
+            ]
+        )
+
+
+class HWEnergyNumberEntity(
+    CoordinatorEntity[HWEnergyDeviceUpdateCoordinator], NumberEntity
+):
+    """Representation of status light number."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the control number."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry.unique_id}_status_light_level"
+        self._attr_name = "Light level"
+        self._attr_native_unit_of_measurement = PERCENTAGE
+        self._attr_icon = "mdi:lightbulb-on"
+        self._attr_device_info = {
+            "name": entry.title,
+            "manufacturer": "HomeWizard",
+            "sw_version": coordinator.data["device"].firmware_version,
+            "model": coordinator.data["device"].product_type,
+            "identifiers": {(DOMAIN, coordinator.data["device"].serial)},
+        }
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set a new value."""
+        await self.coordinator.api.state_set(brightness=value * (255 / 100))
```

Comment:
Why 255 ? You don’t set a native min and a max value.

Reply:
The API's internal level is 0-255, but I think it's more intuitive for the user to make it a percentage, hence the mapping.

Native min/max defaults to 0/100, so it is not set. But I can add this to make it clearer?

Reply:
I think it makes sense as is right now 👍